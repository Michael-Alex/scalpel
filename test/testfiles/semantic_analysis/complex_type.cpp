const int unsigned long* const volatile** const**** i, j;
