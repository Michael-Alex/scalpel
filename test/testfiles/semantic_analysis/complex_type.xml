<namespace id="n0">
 <variable name="i">
  <type>
   <pointer>
    <pointer>
     <pointer>
      <pointer>
       <cv_qualifiers const="true">
        <pointer>
         <pointer>
          <cv_qualifiers const="true" volatile="true">
           <pointer>
            <cv_qualifiers const="true">
             <fundamental_type type="unsigned long int"/>
            </cv_qualifiers>
           </pointer>
          </cv_qualifiers>
         </pointer>
        </pointer>
       </cv_qualifiers>
      </pointer>
     </pointer>
    </pointer>
   </pointer>
  </type>
 </variable>
 <variable name="j">
  <type>
   <cv_qualifiers const="true">
    <fundamental_type type="unsigned long int"/>
   </cv_qualifiers>
  </type>
 </variable>
</namespace>
