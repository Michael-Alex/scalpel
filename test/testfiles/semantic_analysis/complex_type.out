{
  "id": 0,
  "variables":
  [
    {
      "id": 0,
      "name": "i",
      "type":
      {
        "category": "pointer",
        "qualified type":
        {
          "category": "pointer",
          "qualified type":
          {
            "category": "pointer",
            "qualified type":
            {
              "category": "pointer",
              "qualified type":
              {
                "category": "cv-qualified type",
                "const": true,
                "qualified type":
                {
                  "category": "pointer",
                  "qualified type":
                  {
                    "category": "pointer",
                    "qualified type":
                    {
                      "category": "cv-qualified type",
                      "const": true,
                      "volatile": true,
                      "qualified type":
                      {
                        "category": "pointer",
                        "qualified type":
                        {
                          "category": "cv-qualified type",
                          "const": true,
                          "qualified type":
                          {
                            "category": "fundamental type",
                            "type": "unsigned long int"
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    },
    {
      "id": 1,
      "name": "j",
      "type":
      {
        "category": "cv-qualified type",
        "const": true,
        "qualified type":
        {
          "category": "fundamental type",
          "type": "unsigned long int"
        }
      }
    }
  ]
}
